client/me/purchases/manage-purchase/change-payment-method/index.jsx

```diff
@@ -277,6 +277,7 @@ function ChangePaymentMethodList( {
 }
 
 function onChangeComplete( { successCallback, translate, showSuccessMessage } ) {
+	recordTracksEvent( 'calypso_purchases_save_new_payment_method' );
```

Comment:
Because `recordTracksEvent` is a Redux action creator, this call actually just returns an object and that's it. You then have to send that object to the Redux dispatcher. When it's used above as `this.props.recordTracksEvent`, the function has already been passed through the second argument to `connect`, which binds it to the dispatcher automatically, but here you're not inside a component so you can't use `this.props`.

The simplest answer is probably to pass the bound version of the function to `onChangeComplete` in the same way we're passing `translate` (although you'll have to give it a slightly different name to avoid conflicting with the unbound function). So something like:

```js
//...
onChangeComplete( { successCallback, translate, showSuccessMessage, dispatchTracksEvent: this.props.recordTracksEvent } )
//...
function onChangeComplete( { successCallback, translate, showSuccessMessage, dispatchTracksEvent } ) {
  dispatchTracksEvent( 'calypso_purchases_save_new_payment_method' );
```

Reply:
Oh, sorry, `ChangePaymentMethodList` is a functional component, so you need a slightly different technique:

```js
const reduxDispatch = useDispatcher();
//...
onChangeComplete( { successCallback, translate, showSuccessMessage, reduxDispatch } )
//...
function onChangeComplete( { successCallback, translate, showSuccessMessage, reduxDispatch } ) {
	reduxDispatch( recordTracksEvent( 'calypso_purchases_save_new_payment_method' ) );

```

Reply:
yay! That worked. Thanks @sirbrillig!